ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/core/Transaction.java

```diff
@@ -87,14 +89,14 @@ public static Transaction readFrom(final RLPInput input) throws RLPException {
             .value(input.readUInt256Scalar(Wei::wrap))
             .payload(input.readBytesValue());
 
-    final int v = input.readIntScalar();
+    final BigInteger v = input.readBigIntegerScalar();
     final byte recId;
-    int chainId = -1;
-    if (v == REPLAY_UNPROTECTED_V_BASE || v == REPLAY_UNPROTECTED_V_BASE + 1) {
-      recId = (byte) (v - REPLAY_UNPROTECTED_V_BASE);
-    } else if (v > REPLAY_PROTECTED_V_MIN) {
-      chainId = (v - REPLAY_PROTECTED_V_BASE) / 2;
-      recId = (byte) (v - (2 * chainId + REPLAY_PROTECTED_V_BASE));
+    BigInteger chainId = BigInteger.valueOf(-1);
```

Comment:
This probably shouldn't be using `-1` as a sentinel value now that we're not using a primitive.  Probably should be `Optional<BigInteger> chainId = Optional.empty();` and that can flow through to `Builder.chainId`.

Reply:
done